Rn79 migrate UI menu

folarinemurgo opened the pull request:
## Description / Change(s) / Related issue(s)
Completed Menu and Settings Revert

##### Ticket
YV-563


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Expanded Application Settings (network, about, terms, privacy, analytics, Change PIN) and added automatic wallet opener.

* **Refactor**
  * Simplified menu and navigation flows for more consistent routing and behavior.

* **Style**
  * Updated menu/support layouts, spacing, borders and icons for visual consistency.

* **Bug Fixes**
  * Smoother toggle behavior (biometrics, crash reporting, screen sharing), loading states, and prevented repeated modal openings.

* **Documentation**
  * Updated settings localization message IDs.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Opestein committed:
rn79: Migrate-UI-Menu
Added Menu List Options

Opestein committed:
rn79: Migrate-UI-Menu
Added Menu List Options

Opestein committed:
rn79: Migrate-UI-Menu
Reverted Wallet Settings Tab back to implementation

Opestein committed:
rn79: Migrate-UI-Menu
Updated routes

Opestein committed:
rn79: Migrate-UI-Menu
Reverted Applications Settings Tab back to implementation while maintaining new string hooks

Opestein committed:
rn79: Migrate-UI-Menu
Reverted Settings Tab back to implementation while maintaining new string hooks

Opestein committed:
rn79: Migrate-UI-Menu
Updated Imports

Opestein committed:
rn79: Migrate-UI-Menu
Removed Catalyst; hook throwing error

Opestein committed:
rn79: Migrate-UI-Menu
Removed Catalyst; hook throwing error

Opestein committed:
rn79: Migrate UI Menu
Reverted Menu, Wallet and Applications Settings Tab back to implementation while maintaining new string hooks, updated

Opestein committed:
rn79: Migrate-UI-Menu
Updated unsynced files

Opestein committed:
rn79: Migrate-UI-Menu
removed stake navigation

stackchain commented:
@folarinemurgo please fix the conflicts.

Opestein committed:
Merge remote-tracking branch 'origin/develop' into rn79-Migrate-UI-Settings-(Only-UI)

# Conflicts:
#	apps/mobile/src/features/Menu/Menu.tsx
#	apps/mobile/src/features/RegisterCatalyst/common/hooks.ts
#	apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx
#	apps/mobile/src/features/Settings/useCases/changeAppSettings/Currency/ChangeCurrencyScreen.tsx
#	apps/mobile/src/features/Settings/useCases/changeAppSettings/ScreenShare/ScreenShare.ts
#	apps/mobile/src/features/Settings/useCases/changeWalletSettings/ChangePassword/ChangePasswordScreen.tsx
#	apps/mobile/src/features/Settings/useCases/changeWalletSettings/ManageCollateral/InitialCollateralInfoModal.tsx
#	apps/mobile/src/features/Settings/useCases/changeWalletSettings/WalletSettingsScreen.tsx
#	apps/mobile/src/kernel/i18n/messages/menu.ts
#	apps/mobile/src/kernel/i18n/messages/settings.ts

Opestein committed:
rn79: Migrate-UI-Menu
Conflicts resolution with develop

folarinemurgo commented:
> @folarinemurgo please fix the conflicts.

Conflicts resolved

jorbuedo commented on apps/mobile/src/wallets/hooks/index.ts:
This file was purposefully deleted, don't add it back

folarinemurgo replied:
File Removed and Updated

jorbuedo committed:
rn79 Menu/Settings fixes: remove wallets/hooks index; fix Menu item disabled opacity and theme color; ApplicationSettings privacy toggle and screen-sharing loading state; ChangeNetwork modal one-time show + correct strings; ManageCollateral useSelectedWallet and types; WalletSettings useResync hook + confirmation dialog; ThemePickerList simplified to rn71 style

Opestein committed:
rn79: Migrate-UI-Menu
Removed wallet hook

Opestein committed:
Merge remote-tracking branch 'origin/rn79-Migrate-UI-Menu' into rn79-Migrate-UI-Settings-(Only-UI)

stackchain commented on apps/mobile/src/features/RegisterCatalyst/common/hooks.ts:
```diff
@@ -24,35 +24,57 @@ export const useCanVote = (wallet: YoroiWallet) => {
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    const fetchData = async () => {
```
@folarinemurgo no problem in leaving with react-query here, we are migrating away from query what is not fetching, this can stick with rq library

stackchain reviewed (changes requested):
data-fetch can stick with react-query

Opestein committed:
rn79: Migrate-UI-Menu
Updated strings

Opestein committed:
Merge remote-tracking branch 'origin/develop' into rn79-Migrate-UI-Menu

Opestein committed:
rn79: Migrate-UI-Menu
Updated strings

Opestein committed:
rn79: Migrate-UI-Menu
Reverted register catalyst hook to react query usage

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx:
```diff
@@ -22,74 +29,159 @@
     <SafeAreaView
       edges={['bottom', 'right', 'left']}
-      style={[ta.bg_color_max, a.flex_1]}
+      style={[a.flex_row, ta.bg_color_max]}
```
@folarinemurgo this is correct, you need to add wrap it if needed

Opestein committed:
rn79: Migrate-UI-Menu
Updated style

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx:
```diff
@@ -22,17 +29,40 @@ import {useCrashReports} from './useCrashReports'
+
+  const {data: screenShareEnabled, isLoading: isScreenShareLoading} =
+    useScreenShareSettingEnabled()
+  const displayScreenShareSetting = Platform.OS === 'android'
```
this is static, can be moved outside the component, there is a helper for it though, isAndroid from the env/constants

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx:
```diff
@@ -103,19 +195,17 @@
+        setPrivacyModeOff()
+      }
+      return next
+    })
```
this is synchronous now, so the state would reflect the storage as long as the useSyncStorageToState hook is linked

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx:
```diff
@@ -131,28 +221,26 @@ const CrashReportsSwitch = ({
-  const {enable, disable} = useCrashReports()
-  const [isLocalEnabled, setIsLocalEnabled] = React.useState(crashReportEnabled)
+  const {enable, disable, enabled} = useCrashReports()
+  const [isLocalCrashReportEnabled, setIsLocalCrashReportEnabled] =
```
this can be consumed with useSyncStorageToState, then the React state can be dropped

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ApplicationSettingsScreen.tsx:
```diff
@@ -163,39 +251,25 @@ const ScreenSharingSwitch = ({
-  const {changeScreenShareSettings} = useChangeScreenShareSetting()
-  const [isLocalEnabled, setIsLocalEnabled] =
+  const {changeScreenShareSettings, isLoading} = useChangeScreenShareSetting()
+  const [isLocalScreenSharingEnabled, setIsLocalScreenSharingEnabled] =
```
I don't recall if this one was migrate to the new storages (in kernel) if not, copy the pattern for the other properties.

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ChangeNetwork/ChangeNetworkScreen.tsx:
```diff
@@ -16,21 +16,26 @@ import {Platform, Text, View} from 'react-native'
+        // prevent repeated opens during async storage update
+        hasShownRef.current = true
+        handleOpenModal()
+      }, time.seconds(0.5))
```
!error

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ChangeNetwork/ChangeNetworkScreen.tsx:
```diff
@@ -16,21 +16,26 @@
+        // prevent repeated opens during async storage update
+        hasShownRef.current = true
+        handleOpenModal()
+      }, time.seconds(0.5))
```
that's false, yet, needs to be moved to sync storage.

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ChangeTheme/ChangeThemeScreen.tsx:
```diff
@@ -13,7 +13,9 @@ export const ChangeThemeScreen = () => {
     >
-      <ThemePickerList />
+      <Boundary>
+        <ThemePickerList />
```
theme is synchronous now, is the Boundary still needed?

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ChangeTheme/ThemePickerList.tsx:
```diff
@@ -1,17 +1,15 @@
     <FlatList
       contentContainerStyle={{...a.p_lg}}
-      data={Object.entries(supportedThemes).map(([, v]) => ({themeName: v}))}
+      data={Object.entries(supportedThemes).map(([k, v]) => ({themeName: v}))}
```
```suggestion
      data={Object.entries(supportedThemes).map(([, v]) => ({themeName: v}))}
```

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ScreenShare/ScreenShare.ts:
```diff
@@ -11,56 +9,92 @@ const {FlagSecure} = NativeModules
+    }
+  }
 
   return {
```
~return the mutation, keep the react query for now~
migrate to useSyncStorageToState -> can be accomplished with the factory that build a key manager for the storage (check the storages file in the kernel)

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ScreenShare/ScreenShare.ts:
```diff
@@ -11,56 +9,92 @@ const {FlagSecure} = NativeModules
-  return useQuery({
-    queryKey: ['screenShareEnabled'],
-    queryFn: async () => {
+  const fetchScreenShareSetting = async () => {
```
same here

stackchain commented on apps/mobile/src/features/Settings/useCases/changeWalletSettings/ChangePassword/ChangePasswordScreen.tsx:
```diff
@@ -36,6 +35,14 @@ export const ChangePasswordScreen = () => {
+    try {
+      await changePassword({currentPassword, newPassword})
+    } catch (err) {
+      // Error is already handled in the hook
```
this changes the behavior

folarinemurgo replied:
reverted and fixed storage

stackchain commented on apps/mobile/src/features/Settings/useCases/changeWalletSettings/WalletSettingsScreen.tsx:
```diff
@@ -1,250 +1,260 @@
-} from '~/features/Settings/SettingsItems'
-import {useWalletManager} from '~/features/WalletManager/context/WalletManagerProvider'
+import {Wallet} from '@yoroi/types'
+import React from 'react'
```
```suggestion
import * as React from 'react'
```

stackchain reviewed (changes requested):
move legacy screenShare to storages (if it is not there yet), and also the current hooks can be dropped and replaced by the useSyncToStorage -> key maker

Opestein committed:
Merge remote-tracking branch 'origin/develop' into rn79-Migrate-UI-Menu

Opestein committed:
rn79: Migrate UI Menu
Updated hooks

Opestein committed:
Merge remote-tracking branch 'origin/develop' into rn79-Migrate-UI-Menu

Opestein committed:
rn79: Migrate-UI-Menu
fix storing of encrypted.value instead of Hex object in EncryptedStorage

Opestein committed:
rn79: Migrate-UI-Menu
migrate screen share to useSyncStorageToState

folarinemurgo commented:
> move legacy screenShare to storages (if it is not there yet), and also the current hooks can be dropped and replaced by the useSyncToStorage -> key maker

Done

Opestein committed:
rn79: Migrate-UI-Menu
Migrated biometrics to use auth provider instead of useEnableAuthWithOs hook.
Update auth setting to 'pin' when switching from biometrics

stackchain commented on apps/mobile/src/features/Settings/useCases/changeAppSettings/ScreenShare/ScreenShare.ts:
```diff
@@ -1,64 +1,36 @@
   return {
-    ...mutation,
-    changeScreenShareSettings: mutation.mutate,
+    data,
```
this API can be broken can now is sync. return data is fine
